### ISSUE TITLE
Browser distribution: Uncaught TypeError: Cannot read property 'split' of undefined

### ISSUE THREAD

**Issue by cmorten** [rollup/rollup#3934]
- Rollup Version: 2.38.0
- Operating System (or Browser): MacOS
- Node Version (if applicable): Browser
- Link to reproduction (IMPORTANT, read below): <https://gist.github.com/cmorten/fdffbd9ce521b6436f5fcbce429e752d>

### Expected Behavior

BAU, or if an error case arises, a deliberate and descriptive Rollup thrown error (e.g. entry module does not exist) when using the Browser distribution of Rollup.

### Actual Behavior

Executing:

```ts
import {
  rollup,
} from "https://unpkg.com/rollup@2.38.0/dist/es/rollup.browser.js";

const options = {
  input: "src/mod.ts",
};

rollup(options);
```

In the browser (or Deno) results in the following thrown error:

```
error: Uncaught TypeError: Cannot read property 'split' of undefined
    at Et (https://unpkg.com/rollup@2.38.0/dist/es/rollup.browser.js:11:65715)
    at ja (https://unpkg.com/rollup@2.38.0/dist/es/rollup.browser.js:11:325076)
    at async Qa.loadEntryModule (https://unpkg.com/rollup@2.38.0/dist/es/rollup.browser.js:11:334442)
    at async Promise.all (index 0)
```

Please see the reproduction (<https://gist.github.com/cmorten/fdffbd9ce521b6436f5fcbce429e752d>) for complementary details. Summary below:

- Rollup's [browser implementation](https://github.com/rollup/rollup/blob/master/browser/path.ts#L60) of `resolve()` does not cater for no arguments being passed.
- Rollup's [module resolution](https://github.com/rollup/rollup/blob/master/src/utils/resolveId.ts#L31) code calls `resolve()` with no arguments in certain scenarios.
- In NodeJS, calling `resolve()` with no arguments is valid and will return the CWD, but in the Rollup browser implementation results in attempting to call `.split()` on `undefined`.
- I suspect the fix may be as simple as replacing the code at <https://github.com/rollup/rollup/blob/master/browser/path.ts#L61> with

    ```ts
    let resolvedParts = (paths.shift() || "").split(/[/\\]/);
    ```

    but it may well not be that simple. Alternatively, removing the use of `resolve()` with no arguments from <https://github.com/rollup/rollup/blob/master/src/utils/resolveId.ts#L31> with something like

    ```ts
    importer ? resolve(dirname(importer), source) : resolve(source),
    ```

    given that if the provided arguments are insufficient to form an absolute path then the behaviour of `resolve()` is to prepend the CWD, and therefore the inclusion of the `resolve()` currently is both superfluous according to specification, and causing the Rollup browser distribution to thrown an `Uncaught TypeError` in very simple cases.

    > If, after processing all given path segments, an absolute path has not yet been generated, the current working directory is used.

    REF: https://nodejs.org/api/path.html#path_path_resolve_paths

Context: I am using the browser distribution as the core for a [Deno port of Rollup](https://github.com/cmorten/deno-rollup) in which I am having to work around this issue currently.

Happy to put in a PR if the issue stands (and haven't missed something obvious!)

**Comment by shellscape** [comment]
Does this happen with Node LTS? I'm not entirely sure we're supporting Deno as yet.

**Comment by cmorten** [comment]
No this does not happen in Node. It is specifically a problem with the Browser distribution of Rollup which afaik is officially supported (it drives the online REPL for instance afaik) which happens to be the easiest entrypoint also for Deno users.

**Comment by lukastaegert** [comment]
Resolved via #3935
